### PR TITLE
Partial support for Cosmos package list

### PR DESCRIPTION
**Pull request by jsancio**

### CODE REVIEW

File: cli/dcoscli/package/main.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import dcoscli
 import docopt
 import pkg_resources
-from dcos import (cmds, emitting, errors, http, marathon, options, package,
+from dcos import (cmds, emitting, errors, http, options, package,
```
Comment (jsancio) [inline cli/dcoscli/package/main.py]: Wohoo! No more marathon in the package command!

File: dcos/package.py [hunk 2]
```diff
@@ -189,7 +189,8 @@ def installed_packages(init_client, endpoints):
     :rtype: [InstalledPackage]
     """
 
-    apps = installed_apps(init_client, endpoints)
+    # TODO: make sure that endpoints works
```
Comment (jsancio) [inline dcos/package.py]: Filed an issues for this.

File: dcos/package.py [hunk 3]
```diff
@@ -220,65 +221,12 @@ def installed_subcommands():
...
-    return valid_apps
-
-
 def _decode_and_add_context(app_id, labels):
     """ Create an enhanced package JSON from Marathon labels
 
     {
       'appId': <appId>,
       'packageSource': <source>,
-      'registryVersion': <app_version>,
```
Comment (jsancio) [inline dcos/package.py]: We weren't populating this anymore.

File: dcos/package.py [hunk 4]
```diff
@@ -290,6 +238,8 @@ def _decode_and_add_context(app_id, labels):
     :rtype: dict
     """
 
+    # TODO: remove images if it exists
```
Comment (jsancio) [inline dcos/package.py]: Filed an issue for this.

File: dcos/package.py [hunk 5]
```diff
@@ -1991,6 +1941,55 @@ def get_package_version(self, package_name, package_version):
                               pkg.registry(), pkg.get_path())
 
 
+    def installed_apps(package_name, app_id, endpoints=False):
```
Comment (jsancio) [inline dcos/package.py]: We are ignoring package_name and app_id. This is being used by the caller. We need to fix this but it is okay for now.

File: dcos/package.py [hunk 6]
```diff
@@ -2055,5 +2054,53 @@ def get_package_version(self, package_name, package_version):
...
+        """
+
+        params = {}
+        if package_name is not None:
+            params["name"] = package_name
+        if app_id is not None:
+            params["appId"] = app_id
+
+        url = urllib.parse.urljoin(self.cosmos_url, 'package/list')
+        # TODO: handle error cases
```
Comment (jsancio) [inline dcos/package.py]: Filed an issue for this.

File: dcos/subcommand.py [hunk 7]
```diff
@@ -290,8 +290,7 @@ def _subcommand_dir():
                                            constants.DCOS_SUBCOMMAND_SUBDIR))
 
 
-# TODO(mgummelt): should be made private after "dcos subcommand" is removed
-def package_dir(name):
+def _package_dir(name):
```
Comment (jsancio) [inline dcos/subcommand.py]: This TODO was implemented. Just making the necessary changes now.

File: cli/dcoscli/package/main.py (hunk 1 above)
Comment (tamarrow) [inline cli/dcoscli/package/main.py]: :+1:

File: dcos/package.py [hunk 8]
```diff
@@ -2055,5 +2054,53 @@ def get_package_version(self, package_name, package_version):
...
+        :param package_name: the optional package to list
+        :type package_name: str
+        :param app_id: the optional application id to list
+        :type app_id: str
+        :rtype: [dict]
+        """
+
+        params = {}
+        if package_name is not None:
+            params["name"] = package_name
```
Comment (tamarrow) [inline dcos/package.py]: is this `packageName` now?

File: dcos/subcommand.py (hunk 7 above)
Comment (tamarrow) [inline dcos/subcommand.py]: nice

**Comment by tamarrow** [comment]
looks good. should we wait to merge until the list endpoint gets merged in `cosmos`?

**Comment by jsancio** [comment]
Let's merge it @tamarrow. It is going to be a few before cosmos is ready.

On Mon, Jan 25, 2016 at 5:05 PM tamarrow <notifications@github.com> wrote:

> looks good. should we wait to merge until the list endpoint gets merged in
> cosmos?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/dcos-cli/pull/416#issuecomment-174754044>.
>
-- 
José Armando García Sancio
Mesosphere

File: dcos/package.py (hunk 8 above)
Comment (jsancio) [inline dcos/package.py]: Yes. I'll change it right now.

**Comment by tamarrow** [comment]
okay. looks like the syntax tests failed, lets fix those and merge?
```
[00:59:30][Step 4/5] dcos/package.py:1944:5: E303 too many blank lines (2)
[00:59:30][Step 4/5] dcos/package.py:2100:39: E222 multiple spaces after operator
```

**Comment by jsancio** [comment]
@tamarrow Fixed the syntax errors and rebased on top of your changes. I will merge to cosmos for now.

**Pull request merged**